File: elixir/day_2/test/submarine_test.exs
defmodule Day2Test do
  use ExUnit.Case
  doctest Submarine

  test "#calculate_position_and_depth" do
    inputs = [
      {"forward", 5},
      {"down", 5},
      {"forward", 8},
      {"up", 3},
      {"down", 8},
      {"forward", 2}
    ]

    assert Submarine.calculate_position_and_depth(inputs) == %{horizontal: 15, depth: 10}
  end

  test "#advanced_calculate_position_and_depth" do
    inputs = [
      {"forward", 5},
      {"down", 5},
      {"forward", 8},
      {"up", 3},
      {"down", 8},
      {"forward", 2}
    ]

    assert Submarine.advanced_calculate_position_and_depth(inputs) == %{horizontal: 15, depth: 60}
  end
end

File: typescript/src/day_02/__tests__/submarine.test.ts
import { advancedCalculatePositionAndDepth, calculatePositionAndDepth } from '../submarine';

describe('calculatePositionAndDepth', () => {
  it('returns the horizontal position and depth after a planned course', () => {
    const inputs = [
      { 'command': 'forward', 'value': 5},
      { 'command': 'down', 'value': 5},
      { 'command': 'forward', 'value': 8},
      { 'command': 'up', 'value': 3},
      { 'command': 'down', 'value': 8},
      { 'command': 'forward', 'value': 2}
    ]
    expect(calculatePositionAndDepth(inputs)).toEqual({horizontal: 15, depth: 10})
  })
});

describe('advancedCalculatePositionAndDepth', () => {
  it('returns the horizontal position and depth after a planned course', () => {
    const inputs = [
      { 'command': 'forward', 'value': 5},
      { 'command': 'down', 'value': 5},
      { 'command': 'forward', 'value': 8},
      { 'command': 'up', 'value': 3},
      { 'command': 'down', 'value': 8},
      { 'command': 'forward', 'value': 2}
    ]
    expect(advancedCalculatePositionAndDepth(inputs)).toEqual({horizontal: 15, depth: 60})
  })
});
